docs-src/tutorials/04-cookbook.md

```diff
@@ -54,8 +54,15 @@ being show.
 ```javascript
 when: {
   show() {
-    const currentStepElement = shepherd.currentStep.el;
-    const header = currentStepElement.querySelector('.shepherd-header');
+    const currentStep = Shepherd.activeTour?.getCurrentStep()
+    if (!currentStep)
+      return
+    const currentStepElement = currentStep.getElement()
+    if (!currentStepElement)
+      return
+    const header = currentStepElement.querySelector('.shepherd-header')
+    if (!header)
+      return
     const progress = document.createElement('span');
     progress.style['margin-right'] = '315px';
     progress.innerText = `${shepherd.steps.indexOf(shepherd.currentStep) + 1}/${shepherd.steps.length}`;
```

Comment:
```suggestion
    progress.innerText = `${Shepherd.activeTour?.steps.indexOf(currentStep) + 1}/${Shepherd.activeTour?.steps.length}`;
```

Reply:
Not sure if this works or not, but guessing we need to update it?

Reply:
We'll need to do `header?.insertBefore` as well to use optional chaining there, etc. Probably should do a quick pass to make sure we've done that everywhere we can.

Reply:
- added optional chaining to the header/footer in the examples

Is the progress indicator used within the example? I'm not finding step options implicating the show option but I could totally be missing that.

Reply:
@EmNicholson93 no, it is not used in the example. Perhaps we should build a small set of examples that take what we have in the cookbook and actually run it? We could then allow users to see various things they could possibly do.